### ISSUE TITLE
Zest - URL re-encoded when sending requests

### ISSUE THREAD

**Issue by thc202** [zaproxy/zaproxy#3510]
From OWASP ZAP User Group: https://groups.google.com/d/topic/zaproxy-users/XaHmgEX50lk/discussion

Steps to reproduce the issue:
1. Run ZAP with Zest add-on installed;
2. Create a Stand Alone Zest script that sends a request with URL encoded char(s):
```JSON
{
  "about": "",
  "zestVersion": "0.3",
  "title": "URL Re-encoded",
  "description": "",
  "prefix": "",
  "type": "StandAlone",
  "parameters": {
    "tokenStart": "{{",
    "tokenEnd": "}}",
    "tokens": {},
    "elementType": "ZestVariables"
  },
  "statements": [
    {
      "url": "http://example.com/%C3%A7",
      "data": "",
      "method": "GET",
      "headers": "",
      "assertions": [],
      "followRedirects": true,
      "cookies": [],
      "index": 1,
      "enabled": true,
      "elementType": "ZestRequest"
    }
  ],
  "authentication": [],
  "index": 0,
  "enabled": true,
  "elementType": "ZestScript"
}
```
3. Run the Stand Alone script and note that the actual URL sent was encoded again, e.g. `http://example.com/%25C3%25A7`.

Remarks:
If the URL has the char(s) in decoded form (e.g. `http://example.com/ç`) it sends them properly encoded (e.g. `http://example.com/%C3%A7`).
This will require changing the Zest library.

ZAP version:
Version 2.6.0

Add-on:
"Zest - Graphical Security Scripting Language" version 24.

**Comment by thc202** [comment]
@eoftedal what would be your preferred or expected behaviour, never encode and let the "script" (e.g. through another statement) encode it if/when needed?

**Comment by eoftedal** [comment]
@thc202 Yes, never encode, and fail if there are invalid characters like line feeds or spaces in the URL.

**Comment by bshieh** [comment]
Is this items still being worked on? On ZAP 2.9 I'm finding that in zest scripts, the % character is automatically being encoded to %25 by ZAP when the zest script is triggered. Would like it to not encode - just keep it the same. This doesn't happen with context files set with forms-based authentication.

For example, this is used in the zest script: https%3A%2F%2F
Gets encoded to this when ZAP triggers the zest script: https%253A%252F%252F

For this particular authentication flow, the web application gives the wrong response so the authentication flow isn't successful and can't find a workaround.

**Comment by arditmorina** [comment]
Any solution for this bug, I'm still having this bug and I can't find the solution.
